src/SFA.DAS.PR.Application/Requests/Commands/AcceptCreateAccountRequest/AcceptCreateAccountRequestCommandHandler.cs

```diff
@@ -0,0 +1,199 @@
+﻿using System.Text.Json;
+using MediatR;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+using SFA.DAS.Encoding;
+using SFA.DAS.PR.Application.Mediatr.Responses;
+using SFA.DAS.PR.Data;
+using SFA.DAS.PR.Domain.Entities;
+using SFA.DAS.PR.Domain.Interfaces;
+using SFA.DAS.ProviderRelationships.Messages.Events;
+using SFA.DAS.ProviderRelationships.Types.Models;
+
+namespace SFA.DAS.PR.Application.Requests.Commands.AcceptCreateAccountRequest;
+
+public sealed class AcceptCreateAccountRequestCommandHandler(
+    ILogger<AcceptCreateAccountRequestCommandHandler> _logger,
+    IEncodingService _encodingService,
+    IProviderRelationshipsDataContext _providerRelationshipsDataContext,
+    IAccountWriteRepository _accountWriteRepository,
+    IAccountProviderWriteRepository _accountProviderWriteRepository,
+    IAccountLegalEntityWriteRepository _accountLegalEntityWriteRepository,
+    IAccountProviderLegalEntitiesWriteRepository _accountProviderLegalEntitiesWriteRepository,
+    IPermissionsWriteRepository _permissionsWriteRepository,
+    IPermissionsAuditWriteRepository _permissionsAuditWriteRepository,
+    IMessageSession _messageSession,
+    IRequestWriteRepository _requestWriteRepository
+) : IRequestHandler<AcceptCreateAccountRequestCommand, ValidatedResponse<Unit>>
+{
+    public async Task<ValidatedResponse<Unit>> Handle(AcceptCreateAccountRequestCommand command, CancellationToken cancellationToken)
+    {
+        await CreateAccountAndAddPermissions(command, cancellationToken);
+
+        return ValidatedResponse<Unit>.EmptySuccessResponse();
+    }
+
+    private async Task CreateAccountAndAddPermissions(AcceptCreateAccountRequestCommand command, CancellationToken cancellationToken)
+    {
+        Request? request = await _requestWriteRepository.GetRequest(command.RequestId, cancellationToken);
+
+        await CreateAccount(command, cancellationToken);
+
+        await CreateAccountLegalEntity(command, cancellationToken);
+
+        Tuple<bool, AccountProvider> providerResponse = await GetOrCreateAccountProvider(command, request!, cancellationToken);
+            
+        AccountProviderLegalEntity accountProviderLegalEntity = 
```

Comment:
see my notes below about Item1 and Item2.  Shouldn't you be testing the providerResponse is successful before you make this call?